junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -1144,8 +1142,17 @@ public static boolean isNestedClassPresent(Class<?> clazz, Predicate<Class<?>> p
 		Preconditions.notNull(clazz, "Class must not be null");
 		Preconditions.notNull(predicate, "Predicate must not be null");
 
-		boolean visitorWasNotCalled = visitNestedClasses(clazz, predicate, __ -> false);
-		return !visitorWasNotCalled;
+		class Tracker implements Consumer<Class<?>> {
+			boolean foundNestedClass = false;
+			@Override
+			public void accept(Class<?> __) {
+				foundNestedClass = true;
+			}
+		}
+		Tracker tracker = new Tracker();
+
+		visitAllNestedClasses(clazz, predicate, tracker);
+		return tracker.foundNestedClass;
```

Comment:
```suggestion
		AtomicBoolean found = new AtomicBoolean();
		visitAllNestedClasses(clazz, predicate, __ -> found.set(true));
		return found.get();
```

I think I would prefer that we use an `AtomicBoolean` here for simplicity.

Reply:
Ahhh... looks like our comments crossed paths in the ether. 😉